### ISSUE TITLE
Fails to lstat() device mapper /dev/dm-* file

### ISSUE THREAD

**Issue by agunnerson-ibm** [docker/docker#14048]
Description of problem:

When running a container using `docker start`, docker fails with the message: `lstat /dev/dm-5: no such file or directory`. For example, my script runs (with private info redacted):

``` sh
container=$(sudo docker create --privileged=true -v /tmp/jenkins4541145667825072762.jnr:/ssh-agent -e ENV1=val1 registry-01:80/internal/image /init.sh -n -c 'jruby -S bundle exec rake ...' -b production)
sudo docker start "${container}"
...
```

results in:

```
09:25:29 Error response from daemon: Cannot start container b37c6724a46e1819f5aa3cc6bc340ae925d6823edb9bd9c3524a305fb5e0a46d: lstat /dev/dm-5: no such file or directory
09:25:29 time="2015-06-19T09:25:29-04:00" level=fatal msg="Error: failed to start one or more containers" 
```

`docker version`:

```
Client version: 1.6.0
Client API version: 1.18
Go version (client): go1.4.2
Git commit (client): 8aae715/1.6.0
OS/Arch (client): linux/amd64
Server version: 1.6.0
Server API version: 1.18
Go version (server): go1.4.2
Git commit (server): 8aae715/1.6.0
OS/Arch (server): linux/amd64
```

`docker info`:

```
Containers: 1
Images: 57
Storage Driver: devicemapper
 Pool Name: docker-253:0-67663084-pool
 Pool Blocksize: 65.54 kB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 2.356 GB
 Data Space Total: 107.4 GB
 Data Space Available: 49.65 GB
 Metadata Space Used: 4.141 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.143 GB
 Udev Sync Supported: true
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.93-RHEL7 (2015-01-28)
Execution Driver: native-0.2
Kernel Version: 3.10.0-229.4.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
CPUs: 6
Total Memory: 7.634 GiB
Name: centos-slave-01
ID: H5S3:6NKS:XUXZ:ONZV:TIUZ:YYDS:H36Q:PSIH:UPEK:ZO4Y:X5SA:M4PO
```

`uname -a`:

```
Linux centos-slave-01 3.10.0-229.4.2.el7.x86_64 #1 SMP Wed May 13 10:06:09 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

Environment details (AWS, VirtualBox, physical, etc.):

The docker host is running on a HyperV VM on Windows Server 2012 R2.

How reproducible:

So far, this has happened once every 100 or so Jenkins builds (which run a docker image). Up to 4 jobs are run in parallel, so there are usually 3 or 4 docker containers running simultaneously.

Additional info:

Here's what `journalctl` logged around the time of the error +/- 1 second:

```
[centos@centos-slave-01 ~]$ sudo journalctl --since "2015-06-19 09:25:28" --until "2015-06-19 09:25:30" -l --no-pager
-- Logs begin at Thu 2015-06-18 17:01:14 EDT, end at Fri 2015-06-19 11:55:24 EDT. --
Jun 19 09:25:29 centos-slave-01 kernel: EXT4-fs (dm-5): mounted filesystem with ordered data mode. Opts: discard
Jun 19 09:25:29 centos-slave-01 kernel: SELinux: initialized (dev dm-5, type ext4), uses xattr
Jun 19 09:25:29 centos-slave-01 docker[1435]: time="2015-06-19T09:25:29-04:00" level=info msg="+job allocate_interface(b37c6724a46e1819f5aa3cc6bc340ae925d6823edb9bd9c3524a305fb5e0a46d)"
Jun 19 09:25:29 centos-slave-01 kernel: EXT4-fs (dm-6): mounted filesystem with ordered data mode. Opts: discard
Jun 19 09:25:29 centos-slave-01 kernel: SELinux: initialized (dev dm-6, type ext4), uses xattr
Jun 19 09:25:29 centos-slave-01 docker[1435]: time="2015-06-19T09:25:29-04:00" level=info msg="-job allocate_interface(b37c6724a46e1819f5aa3cc6bc340ae925d6823edb9bd9c3524a305fb5e0a46d) = OK (0)"
Jun 19 09:25:29 centos-slave-01 kernel: SELinux: initialized (dev tmpfs, type tmpfs), uses transition SIDs
Jun 19 09:25:29 centos-slave-01 docker[1435]: time="2015-06-19T09:25:29-04:00" level=info msg="+job log(start, b37c6724a46e1819f5aa3cc6bc340ae925d6823edb9bd9c3524a305fb5e0a46d, registry-01:80/internal/image:latest)"
Jun 19 09:25:29 centos-slave-01 docker[1435]: time="2015-06-19T09:25:29-04:00" level=info msg="-job log(start, b37c6724a46e1819f5aa3cc6bc340ae925d6823edb9bd9c3524a305fb5e0a46d, registry-01:80/internal/image:latest) = OK (0)"
Jun 19 09:25:29 centos-slave-01 docker[1435]: time="2015-06-19T09:25:29-04:00" level=info msg="+job release_interface(b37c6724a46e1819f5aa3cc6bc340ae925d6823edb9bd9c3524a305fb5e0a46d)"
Jun 19 09:25:29 centos-slave-01 docker[1435]: time="2015-06-19T09:25:29-04:00" level=info msg="-job release_interface(b37c6724a46e1819f5aa3cc6bc340ae925d6823edb9bd9c3524a305fb5e0a46d) = OK (0)"
Jun 19 09:25:29 centos-slave-01 docker[1435]: time="2015-06-19T09:25:29-04:00" level=info msg="+job log(create, c032ad5cae96e6930ee03b703dc26e57767f4ed43a71a7e73c43c15a2af8c315, registry-01:80/internal/image:latest)"
Jun 19 09:25:29 centos-slave-01 docker[1435]: time="2015-06-19T09:25:29-04:00" level=info msg="-job log(create, c032ad5cae96e6930ee03b703dc26e57767f4ed43a71a7e73c43c15a2af8c315, registry-01:80/internal/image:latest) = OK (0)"
Jun 19 09:25:29 centos-slave-01 docker[1435]: time="2015-06-19T09:25:29-04:00" level=info msg="-job create() = OK (0)"
Jun 19 09:25:29 centos-slave-01 sudo[49237]: jenkins : TTY=unknown ; PWD=/home/jenkins/workspace/rspec_prod_monitors_perf_mgmt_cd_3411 ; USER=root ; COMMAND=/bin/docker start c032ad5cae96e6930ee03b703dc26e57767f4ed43a71a7e73c43c15a2af8c315
Jun 19 09:25:29 centos-slave-01 docker[1435]: time="2015-06-19T09:25:29-04:00" level=info msg="POST /v1.18/containers/c032ad5cae96e6930ee03b703dc26e57767f4ed43a71a7e73c43c15a2af8c315/start"
Jun 19 09:25:29 centos-slave-01 docker[1435]: time="2015-06-19T09:25:29-04:00" level=info msg="+job start(c032ad5cae96e6930ee03b703dc26e57767f4ed43a71a7e73c43c15a2af8c315)"
Jun 19 09:25:29 centos-slave-01 docker[1435]: time="2015-06-19T09:25:29-04:00" level=info msg="+job release_interface(b37c6724a46e1819f5aa3cc6bc340ae925d6823edb9bd9c3524a305fb5e0a46d)"
Jun 19 09:25:29 centos-slave-01 docker[1435]: time="2015-06-19T09:25:29-04:00" level=info msg="-job release_interface(b37c6724a46e1819f5aa3cc6bc340ae925d6823edb9bd9c3524a305fb5e0a46d) = OK (0)"
Jun 19 09:25:29 centos-slave-01 docker[1435]: time="2015-06-19T09:25:29-04:00" level=info msg="+job allocate_interface(c032ad5cae96e6930ee03b703dc26e57767f4ed43a71a7e73c43c15a2af8c315)"
Jun 19 09:25:29 centos-slave-01 docker[1435]: time="2015-06-19T09:25:29-04:00" level=error msg="Error unmounting device b37c6724a46e1819f5aa3cc6bc340ae925d6823edb9bd9c3524a305fb5e0a46d: UnmountDevice: device not-mounted id b37c6724a46e1819f5aa3cc6bc340ae925d6823edb9bd9c3524a305fb5e0a46d"
Jun 19 09:25:29 centos-slave-01 docker[1435]: time="2015-06-19T09:25:29-04:00" level=info msg="+job log(die, b37c6724a46e1819f5aa3cc6bc340ae925d6823edb9bd9c3524a305fb5e0a46d, registry-01:80/internal/image:latest)"
Jun 19 09:25:29 centos-slave-01 docker[1435]: time="2015-06-19T09:25:29-04:00" level=info msg="-job log(die, b37c6724a46e1819f5aa3cc6bc340ae925d6823edb9bd9c3524a305fb5e0a46d, registry-01:80/internal/image:latest) = OK (0)"
Jun 19 09:25:29 centos-slave-01 docker[1435]: Cannot start container b37c6724a46e1819f5aa3cc6bc340ae925d6823edb9bd9c3524a305fb5e0a46d: lstat /dev/dm-5: no such file or directory
Jun 19 09:25:29 centos-slave-01 docker[1435]: time="2015-06-19T09:25:29-04:00" level=info msg="-job start(b37c6724a46e1819f5aa3cc6bc340ae925d6823edb9bd9c3524a305fb5e0a46d) = ERR (1)"
Jun 19 09:25:29 centos-slave-01 docker[1435]: time="2015-06-19T09:25:29-04:00" level=error msg="Handler for POST /containers/{name:.*}/start returned error: Cannot start container b37c6724a46e1819f5aa3cc6bc340ae925d6823edb9bd9c3524a305fb5e0a46d: lstat /dev/dm-5: no such file or directory"
Jun 19 09:25:29 centos-slave-01 docker[1435]: time="2015-06-19T09:25:29-04:00" level=error msg="HTTP Error: statusCode=404 Cannot start container b37c6724a46e1819f5aa3cc6bc340ae925d6823edb9bd9c3524a305fb5e0a46d: lstat /dev/dm-5: no such file or directory"
Jun 19 09:25:29 centos-slave-01 kernel: EXT4-fs (dm-5): mounted filesystem with ordered data mode. Opts: discard
Jun 19 09:25:29 centos-slave-01 docker[1435]: time="2015-06-19T09:25:29-04:00" level=info msg="-job allocate_interface(c032ad5cae96e6930ee03b703dc26e57767f4ed43a71a7e73c43c15a2af8c315) = OK (0)"
Jun 19 09:25:29 centos-slave-01 kernel: SELinux: initialized (dev dm-5, type ext4), uses xattr
Jun 19 09:25:29 centos-slave-01 kernel: SELinux: initialized (dev tmpfs, type tmpfs), uses transition SIDs
Jun 19 09:25:29 centos-slave-01 docker[1435]: time="2015-06-19T09:25:29-04:00" level=info msg="+job log(start, c032ad5cae96e6930ee03b703dc26e57767f4ed43a71a7e73c43c15a2af8c315, registry-01:80/internal/image:latest)"
Jun 19 09:25:29 centos-slave-01 docker[1435]: time="2015-06-19T09:25:29-04:00" level=info msg="-job log(start, c032ad5cae96e6930ee03b703dc26e57767f4ed43a71a7e73c43c15a2af8c315, registry-01:80/internal/image:latest) = OK (0)"
Jun 19 09:25:29 centos-slave-01 sudo[49253]: jenkins : TTY=unknown ; PWD=/home/jenkins/workspace/rspec_prod_monitors_cd_3411 ; USER=root ; COMMAND=/bin/docker cp b37c6724a46e1819f5aa3cc6bc340ae925d6823edb9bd9c3524a305fb5e0a46d:/myapp/reports/ -
Jun 19 09:25:29 centos-slave-01 kernel: SELinux: initialized (dev mqueue, type mqueue), uses transition SIDs
Jun 19 09:25:29 centos-slave-01 kernel: SELinux: initialized (dev proc, type proc), uses genfs_contexts
Jun 19 09:25:29 centos-slave-01 systemd[1]: Starting docker container c032ad5cae96e6930ee03b703dc26e57767f4ed43a71a7e73c43c15a2af8c315.
Jun 19 09:25:29 centos-slave-01 systemd[1]: Started docker container c032ad5cae96e6930ee03b703dc26e57767f4ed43a71a7e73c43c15a2af8c315.
Jun 19 09:25:29 centos-slave-01 kernel: device veth5cdde34 entered promiscuous mode
Jun 19 09:25:29 centos-slave-01 kernel: IPv6: ADDRCONF(NETDEV_UP): veth5cdde34: link is not ready
Jun 19 09:25:29 centos-slave-01 docker[1435]: time="2015-06-19T09:25:29-04:00" level=info msg="POST /v1.18/containers/b37c6724a46e1819f5aa3cc6bc340ae925d6823edb9bd9c3524a305fb5e0a46d/copy"
Jun 19 09:25:29 centos-slave-01 docker[1435]: time="2015-06-19T09:25:29-04:00" level=info msg="+job container_copy(b37c6724a46e1819f5aa3cc6bc340ae925d6823edb9bd9c3524a305fb5e0a46d, myapp/reports/)"
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>  (veth7f54f3a): carrier is OFF
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>  (veth7f54f3a): new Veth device (driver: 'unknown' ifindex: 1086)
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>  (veth7f54f3a): exported as /org/freedesktop/NetworkManager/Devices/862
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>  (veth5cdde34): carrier is OFF
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>  (veth5cdde34): new Veth device (driver: 'unknown' ifindex: 1087)
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>  (veth5cdde34): exported as /org/freedesktop/NetworkManager/Devices/863
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>  (docker0): bridge port veth5cdde34 was attached
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>  (veth5cdde34): enslaved to docker0
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>      read connection 'veth5cdde34'
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>  (veth5cdde34): device state change: unmanaged -> unavailable (reason 'connection-assumed') [10 20 41]
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>  (veth5cdde34): device state change: unavailable -> disconnected (reason 'connection-assumed') [20 30 41]
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>  (veth5cdde34): Activation: starting connection 'veth5cdde34'
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>  (veth5cdde34): Activation: Stage 1 of 5 (Device Prepare) scheduled...
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>  (veth5cdde34): Activation: Stage 1 of 5 (Device Prepare) started...
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>  (veth5cdde34): device state change: disconnected -> prepare (reason 'none') [30 40 0]
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>  (veth5cdde34): Activation: Stage 2 of 5 (Device Configure) scheduled...
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>  (veth5cdde34): Activation: Stage 1 of 5 (Device Prepare) complete.
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>  (veth5cdde34): Activation: Stage 2 of 5 (Device Configure) starting...
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>  (veth5cdde34): device state change: prepare -> config (reason 'none') [40 50 0]
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>  (veth5cdde34): Activation: Stage 2 of 5 (Device Configure) successful.
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>  (veth5cdde34): Activation: Stage 3 of 5 (IP Configure Start) scheduled.
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>  (veth5cdde34): Activation: Stage 2 of 5 (Device Configure) complete.
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>  (veth5cdde34): Activation: Stage 3 of 5 (IP Configure Start) started...
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>  (veth5cdde34): device state change: config -> ip-config (reason 'none') [50 70 0]
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>  (veth5cdde34): Activation: Stage 3 of 5 (IP Configure Start) complete.
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>  (veth5cdde34): device state change: ip-config -> secondaries (reason 'none') [70 90 0]
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>  (veth5cdde34): device state change: secondaries -> activated (reason 'none') [90 100 0]
Jun 19 09:25:29 centos-slave-01 kernel: IPv6: ADDRCONF(NETDEV_CHANGE): veth5cdde34: link becomes ready
Jun 19 09:25:29 centos-slave-01 kernel: docker0: port 2(veth5cdde34) entered forwarding state
Jun 19 09:25:29 centos-slave-01 kernel: docker0: port 2(veth5cdde34) entered forwarding state
Jun 19 09:25:29 centos-slave-01 kernel: SELinux: initialized (dev tmpfs, type tmpfs), uses transition SIDs
Jun 19 09:25:29 centos-slave-01 kernel: SELinux: initialized (dev devpts, type devpts), uses transition SIDs
Jun 19 09:25:29 centos-slave-01 kernel: SELinux: initialized (dev tmpfs, type tmpfs), uses transition SIDs
Jun 19 09:25:29 centos-slave-01 kernel: SELinux: initialized (dev sysfs, type sysfs), uses genfs_contexts
Jun 19 09:25:29 centos-slave-01 kernel: EXT4-fs (dm-6): mounted filesystem with ordered data mode. Opts: discard
Jun 19 09:25:29 centos-slave-01 kernel: SELinux: initialized (dev dm-6, type ext4), uses xattr
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>  (veth5cdde34): Activation: successful, device activated.
Jun 19 09:25:29 centos-slave-01 dbus-daemon[816]: dbus[816]: [system] Activating via systemd: service name='org.freedesktop.nm_dispatcher' unit='dbus-org.freedesktop.nm-dispatcher.service'
Jun 19 09:25:29 centos-slave-01 dbus[816]: [system] Activating via systemd: service name='org.freedesktop.nm_dispatcher' unit='dbus-org.freedesktop.nm-dispatcher.service'
Jun 19 09:25:29 centos-slave-01 systemd[1]: Starting Network Manager Script Dispatcher Service...
Jun 19 09:25:29 centos-slave-01 NetworkManager[909]: <info>  (veth5cdde34): link connected
Jun 19 09:25:29 centos-slave-01 dbus-daemon[816]: dbus[816]: [system] Successfully activated service 'org.freedesktop.nm_dispatcher'
Jun 19 09:25:29 centos-slave-01 dbus[816]: [system] Successfully activated service 'org.freedesktop.nm_dispatcher'
Jun 19 09:25:29 centos-slave-01 systemd[1]: Started Network Manager Script Dispatcher Service.
Jun 19 09:25:29 centos-slave-01 nm-dispatcher[49280]: Dispatching action 'up' for veth5cdde34
[centos@centos-slave-01 ~]$
```

EDIT: Another person who seems to be having the same issue: http://stackoverflow.com/questions/29950967/docker-fails-to-start-container-occasionally-with-error-lstat-dev-dm-1-no-suc

**Comment by rhvgoyal** [comment]
Hm..., not sure who is calling lstat() here on dm-5 device? Are there any debug logs too? It might help to enable debugging on daemon (-D) and let it run and see if additional logs show up at the time of error.

**Comment by rhvgoyal** [comment]
Following seems to be one odd error.

Jun 19 09:25:29 centos-slave-01 docker[1435]: time="2015-06-19T09:25:29-04:00" level=error msg="Error unmounting device b37c6724a46e1819f5aa3cc6bc340ae925d6823edb9bd9c3524a305fb5e0a46d: UnmountDevice: device not-mounted id b37c6724a46e1819f5aa3cc6bc340ae925d6823edb9bd9c3524a305fb5e0a46d"

Looks like we called Put() one a device where we had never called a Get(). Not sure how did we do that.

**Comment by lsm5** [comment]
@andrewgunnersonexplorys hi, what's `rpm -qi docker` saying for you?

**Comment by andrewgunnersonexplorys** [comment]
@lsm5 Here the docker RPM version:

```
[centos@cd-ops-01 ~]$ rpm -qi docker
Name        : docker
Version     : 1.6.0
Release     : 11.0.1.el7.centos
Architecture: x86_64
Install Date: Thu 04 Jun 2015 02:35:08 PM EDT
Group       : Unspecified
Size        : 33835427
License     : ASL 2.0
Signature   : RSA/SHA256, Wed 13 May 2015 07:50:02 PM EDT, Key ID 24c6a8a7f4a80eb5
Source RPM  : docker-1.6.0-11.0.1.el7.centos.src.rpm
Build Date  : Wed 13 May 2015 07:47:06 PM EDT
Build Host  : worker1.bsys.centos.org
Relocations : (not relocatable)
Packager    : CentOS BuildSystem <http://bugs.centos.org>
Vendor      : CentOS
URL         : http://www.docker.com
Summary     : Automates deployment of containerized applications
Description :
Docker is an open-source engine that automates the deployment of any
application as a lightweight, portable, self-sufficient container that will
run virtually anywhere.

Docker containers can encapsulate any payload, and will run consistently on
and between virtually any server. The same container that a developer builds
and tests on a laptop will run at scale, in production*, on VMs, bare-metal
servers, OpenStack clusters, public instances, or combinations of the above.
[centos@cd-ops-01 ~]$
```

**Comment by thaJeztah** [comment]
@rhvgoyal @lsm5 any status update here? I see this is on an older version of docker (1.6.0); have there been changes in this area since?

**Comment by rhvgoyal** [comment]
@thaJeztah 

No progress on this. I have not seen this and if reporter can reproduce it then we can make some progress. May be close this and reporter can reopen this again when problem can be reproduced.

**Comment by thaJeztah** [comment]
Alright, @agunnerson-explorys I'm going to close this issue for now, because unfortunately, we don't have a way to reproduce this. If you have additional information that can assist in reproducing this, please comment here then we can reopen. Thanks!

**Comment by agunnerson-explorys** [comment]
@thaJeztah Thanks! We have since upgraded to docker 1.7.1 on all our servers and have not run into this issue again after more than 10000 jenkins builds (which spawn short-lived containers).

**Comment by thaJeztah** [comment]
Good to hear, @agunnerson-explorys !

**Comment by litmanovich** [comment]
We have docker 1.7.1 and still got this problem:

11:40:10 Error response from daemon: Cannot start container 96881bb6d7a6afd6e6d448a50bcfe37a8c2f10a1f8e7b9407a2bc13abe4f5abd: lstat /dev/dm-6: no such file or directory

**Comment by thaJeztah** [comment]
@litmanovich could you give the output of `docker info`, `docker version` and `uname -a` ?

**Comment by litmanovich** [comment]
> docker info
Containers: 34
Images: 159
Storage Driver: devicemapper
 Pool Name: docker-253:1-1076717665-pool
 Pool Blocksize: 65.54 kB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 25.64 GB
 Data Space Total: 107.4 GB
 Data Space Available: 81.74 GB
 Metadata Space Used: 25.25 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.122 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.93-RHEL7 (2015-01-28)
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.10.0-229.7.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
CPUs: 16
Total Memory: 31.26 GiB
Name: lego.xiolab.lab.emc.com
ID: VPUZ:HQIB:IQA5:IFYO:6AHR:DSWK:YU2L:IJBF:T333:4GN6:ZU43:ZKBQ

> docker version
Client version: 1.7.1
Client API version: 1.19
Go version (client): go1.4.2
Git commit (client): 786b29d
OS/Arch (client): linux/amd64
Server version: 1.7.1
Server API version: 1.19
Go version (server): go1.4.2
Git commit (server): 786b29d
OS/Arch (server): linux/amd64

> uname -a
Linux lego.xiolab.lab.emc.com 3.10.0-229.7.2.el7.x86_64 #1 SMP Tue Jun 23 22:06:11 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

**Comment by thaJeztah** [comment]
Thanks, @litmanovich. I'll reopen this issue and will ask @rhvgoyal if there's other info that could help in locating this issue. Is this a machine that can be kept in this state for debugging, or do you rely on it for production?

ping @rhvgoyal could you have a look here?

**Comment by litmanovich** [comment]
This is a test machine which runs continuous builds to make sure the environment is stable. I think we can keep it in this state for a while.

**Comment by thaJeztah** [comment]
@litmanovich thanks! let me see if I can get some attention here :-)

ping @vbatts @rhvgoyal could you have a look please?

**Comment by rhvgoyal** [comment]
@litmanovich 

Can you enable debug messages in docker daemon using -D option and try to start container again and see if there are any additional debug messages in logs.

Right now I can't even figure out where is the lstat() call in the code. I need to nail that down first.

**Comment by litmanovich** [comment]
I added -D to daemon options, will send the log when it fails again with lstat error

**Comment by rhvgoyal** [comment]
@litmanovich 

Is that container "96881bb6d7a6afd6e6d448a50bcfe37a8c2f10a1f8e7b9407a2bc13abe4f5abd" still around? Can we try restarting same container and see if problem happens or not.

**Comment by litmanovich** [comment]
No, we have to delete the containers, otherwise after a certain number of exited containers docker gets  stuck.

**Comment by rhvgoyal** [comment]
@litmanovich 

Ok, so if this time you get into this error, keep that container around. We will have to start it multiple times in an attempt to be able to reproduce the problem and take it from there.

**Comment by litmanovich** [comment]
Ok, I will keep the container if it fails with lstat error.

**Comment by alp26** [comment]
We're experiencing the same problem:

Execute command failed with reason:
06:02:30.786 time="2016-01-26T06:02:11Z" level="fatal" msg="Error response from daemon: Cannot start container 7b78bf7ac1984206b3420c3e048356a39358ef0ebd9861e39f72f03ab5012738: lstat /dev/dm-16: no such file or directory" 

docker info
=========
Containers: 9
Images: 1637
Storage Driver: devicemapper
 Pool Name: docker-202:1-263458-pool
 Pool Blocksize: 65.54 kB
 Backing Filesystem: extfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 85.53 GB
 Data Space Total: 429.5 GB
 Data Space Available: 344 GB
 Metadata Space Used: 128.4 MB
 Metadata Space Total: 10.74 GB
 Metadata Space Available: 10.61 GB
 Udev Sync Supported: true
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.89-RHEL6 (2014-09-01)
Execution Driver: native-0.2
Kernel Version: 3.14.48-33.39.amzn1.x86_64
Operating System: Amazon Linux AMI 2015.03
CPUs: 8
Total Memory: 14.69 GiB
Name: ip-172-31-0-148.us-west-2.compute.internal
ID: UHPT:MRR2:NPRA:J42N:E2PC:I3DR:C2JC:F2RU:KGWW:HENQ:MHB3:P3H4


docker version
===========
Client version: 1.6.2
Client API version: 1.18
Go version (client): go1.3.3
Git commit (client): 7c8fca2/1.6.2
OS/Arch (client): linux/amd64
Server version: 1.6.2
Server API version: 1.18
Go version (server): go1.3.3
Git commit (server): 7c8fca2/1.6.2
OS/Arch (server): linux/amd64

uname -a
========
Linux ip-172-31-0-148.us-west-2.compute.internal 3.14.48-33.39.amzn1.x86_64 #1 SMP Tue Jul 14 23:43:07 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

**Comment by hynek1** [comment]
@rhvgoyal We have the same issue - it looks to me as if the unmount at the time of container creation (??) is attempted twice and the second try fails - this happens with layered containers when we start many at the same time. 

relevant part of docker -D log file: (long hex number is replaced by HASH for better readability)

`time="2016-03-18T14:42:11.119517904+01:00" level=debug msg="[deviceset] AddDevice(hash=HASH-init basehash=255e8be8ad19efa00a4da8eabf0600e2e67d38b733212be0c681b4d48fcf4f81)"
time="2016-03-18T14:42:31.385505943+01:00" level=debug msg="registerDevice(1727, HASH-init)"
time="2016-03-18T14:42:31.386673845+01:00" level=debug msg="[deviceset] AddDevice(hash=HASH-init basehash=255e8be8ad19efa00a4da8eabf0600e2e67d38b733212be0c681b4d48fcf4f81) END"
time="2016-03-18T14:42:31.386808090+01:00" level=debug msg="activateDeviceIfNeeded(HASH-init)"
time="2016-03-18T14:42:31.476980439+01:00" level=debug msg="[deviceset] AddDevice(hash=HASH basehash=HASH-init)"
time="2016-03-18T14:42:32.843719711+01:00" level=debug msg="registerDevice(1730, HASH)"
time="2016-03-18T14:42:32.847157632+01:00" level=debug msg="[deviceset] AddDevice(hash=HASH basehash=HASH-init) END"
time="2016-03-18T14:42:32.847273812+01:00" level=debug msg="[devmapper] UnmountDevice(hash=HASH-init)"
time="2016-03-18T14:42:32.847400768+01:00" level=debug msg="[devmapper] Unmount(/var/XXXXXX/docker/devicemapper/mnt/HASH-init)"
time="2016-03-18T14:42:32.852467773+01:00" level=debug msg="[devmapper] deactivateDevice(HASH-init)"
time="2016-03-18T14:42:32.852683078+01:00" level=debug msg="[devmapper] removeDevice START(docker-202:1-1182467-HASH-init)"
time="2016-03-18T14:42:32.873361239+01:00" level=debug msg="[devmapper] removeDevice END(docker-202:1-1182467-HASH-init)"
time="2016-03-18T14:42:32.873434406+01:00" level=debug msg="[devmapper] deactivateDevice END(HASH-init)"
time="2016-03-18T14:42:32.873478179+01:00" level=debug msg="[devmapper] UnmountDevice(hash=HASH-init) END"
time="2016-03-18T14:42:32.873940295+01:00" level=debug msg="activateDeviceIfNeeded(HASH)"
time="2016-03-18T14:42:33.129493665+01:00" level=debug msg="[devmapper] UnmountDevice(hash=HASH)"
time="2016-03-18T14:42:33.645445409+01:00" level=debug msg="[devmapper] Unmount(/var/XXXXXX/docker/devicemapper/mnt/HASH)"
time="2016-03-18T14:42:33.646738110+01:00" level=debug msg="[devmapper] deactivateDevice(HASH)"
time="2016-03-18T14:42:33.646895729+01:00" level=debug msg="[devmapper] removeDevice START(docker-202:1-1182467-HASH)"
time="2016-03-18T14:42:33.679600654+01:00" level=debug msg="[devmapper] removeDevice END(docker-202:1-1182467-HASH)"
time="2016-03-18T14:42:33.679773973+01:00" level=debug msg="[devmapper] deactivateDevice END(HASH)"
time="2016-03-18T14:42:33.679896842+01:00" level=debug msg="[devmapper] UnmountDevice(hash=HASH) END"
time="2016-03-18T14:42:33.683164266+01:00" level=info msg="POST /v1.19/containers/HASH/attach?stderr=1&stdout=1&stream=1"
time="2016-03-18T14:42:33.684436695+01:00" level=info msg="POST /v1.19/containers/HASH/start"
time="2016-03-18T14:42:34.703045315+01:00" level=debug msg="activateDeviceIfNeeded(HASH)"
time="2016-03-18T14:42:35.754121947+01:00" level=debug msg="[devmapper] UnmountDevice(hash=HASH)"
time="2016-03-18T14:42:36.576109604+01:00" level=debug msg="[devmapper] Unmount(/var/XXXXXX/docker/devicemapper/mnt/HASH)"
time="2016-03-18T14:42:36.577234357+01:00" level=debug msg="[devmapper] deactivateDevice(HASH)"
time="2016-03-18T14:42:36.577351046+01:00" level=debug msg="[devmapper] removeDevice START(docker-202:1-1182467-HASH)"
time="2016-03-18T14:42:36.593407181+01:00" level=debug msg="[devmapper] removeDevice END(docker-202:1-1182467-HASH)"
time="2016-03-18T14:42:36.593499538+01:00" level=debug msg="[devmapper] deactivateDevice END(HASH)"
time="2016-03-18T14:42:36.593549457+01:00" level=debug msg="[devmapper] UnmountDevice(hash=HASH) END"
time="2016-03-18T14:42:36.595410468+01:00" level=debug msg="[devmapper] UnmountDevice(hash=HASH)"
time="2016-03-18T14:42:37.322899230+01:00" level=debug msg="[devmapper] UnmountDevice(hash=HASH) END"
time="2016-03-18T14:42:37.323002369+01:00" level=error msg="Error unmounting device HASH: UnmountDevice: device not-mounted id HASH"
time="2016-03-18T14:42:37.323235901+01:00" level=error msg="Handler for POST /containers/{name:.*}/start returned error: Cannot start container HASH: lstat /dev/dm-5: no such file or directory"
time="2016-03-18T14:42:37.323331770+01:00" level=error msg="HTTP Error" err="Cannot start container HASH: lstat /dev/dm-5: no such file or directory" statusCode=404
time="2016-03-18T14:42:37.325478312+01:00" level=info msg="DELETE /v1.19/containers/HASH?v=1"
time="2016-03-18T14:42:39.272172635+01:00" level=debug msg="unregisterDevice(1730, HASH)"
time="2016-03-18T14:42:39.276613998+01:00" level=debug msg="unregisterDevice(1727, HASH-init)"
`

docker 1.7.1 from centos 6, kernel 2.6.32-573.18.1.el6.x86_64, docker fs is devmapper over two block devices


# docker info
Containers: 124
Images: 25
Storage Driver: devicemapper
 Pool Name: docker-202:1-1182467-pool
 Pool Blocksize: 65.54 kB
 Backing Filesystem: extfs
 Data file: /dev/xvdb
 Metadata file: /dev/xvdc
 Data Space Used: 1.687 GB
 Data Space Total: 21.47 GB
 Data Space Available: 19.79 GB
 Metadata Space Used: 3.412 MB
 Metadata Space Total: 1.074 GB
 Metadata Space Available: 1.07 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Library Version: 1.02.95-RHEL6 (2015-09-08)
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 2.6.32-573.18.1.el6.x86_64
Operating System: <unknown>
CPUs: 4
Total Memory: 1.825 GiB
Name: test
ID: 4X4D:C6WM:W5GC:NSYO:W6YS:T72M:ZEZW:QLFG:A672:UWAO:MG5A:QG56
Debug mode (server): true
File Descriptors: 116
Goroutines: 123
System Time: 2016-03-18T14:42:32.613107842+01:00
EventsListeners: 0
Init SHA1: d7be48329e05108a06773d2930d7726012152fe6
Init Path: /usr/libexec/docker/dockerinit
Docker Root Dir: /var/XXXXX/docker

# docker version
Client version: 1.7.1
Client API version: 1.19
Go version (client): go1.4.2
Git commit (client): 786b29d/1.7.1
OS/Arch (client): linux/amd64
Server version: 1.7.1
Server API version: 1.19
Go version (server): go1.4.2
Git commit (server): 786b29d/1.7.1
OS/Arch (server): linux/amd64

**Comment by rhvgoyal** [comment]
You are running on rhel6 kernel and that's not supported with docker.

**Comment by hynek1** [comment]
I'm getting the same on 4.5.0-1.el6.elrepo.x86_64 and 3.10.101-1.el6.elrepo.x86_64 too. 

What kernel version do you recommend?

### docker -D output for 3.10.101-1

```
time="2016-03-29T14:28:45.828975559+02:00" level=debug msg="[deviceset] AddDevice(hash=HASH-init basehash=255e8be8ad19efa00a4da8eabf0600e2e67d38b733212be0c681b4d48fcf4f81)" 
time="2016-03-29T14:29:11.257143830+02:00" level=debug msg="registerDevice(6271, HASH-init)" 
time="2016-03-29T14:29:11.259414459+02:00" level=debug msg="[deviceset] AddDevice(hash=HASH-init basehash=255e8be8ad19efa00a4da8eabf0600e2e67d38b733212be0c681b4d48fcf4f81) END" 
time="2016-03-29T14:29:11.259528404+02:00" level=debug msg="activateDeviceIfNeeded(HASH-init)" 
time="2016-03-29T14:29:11.298645982+02:00" level=debug msg="[deviceset] AddDevice(hash=HASH basehash=HASH-init)" 
time="2016-03-29T14:29:11.950701419+02:00" level=debug msg="registerDevice(6273, HASH)" 
time="2016-03-29T14:29:11.952815557+02:00" level=debug msg="[deviceset] AddDevice(hash=HASH basehash=HASH-init) END" 
time="2016-03-29T14:29:11.953396514+02:00" level=debug msg="[devmapper] UnmountDevice(hash=HASH-init)" 
time="2016-03-29T14:29:12.191302178+02:00" level=debug msg="[devmapper] Unmount(/var/XXXX/docker/devicemapper/mnt/HASH-init)" 
time="2016-03-29T14:29:12.192679037+02:00" level=debug msg="[devmapper] deactivateDevice(HASH-init)" 
time="2016-03-29T14:29:12.192818511+02:00" level=debug msg="[devmapper] removeDevice START(docker-202:1-1182467-HASH-init)" 
time="2016-03-29T14:29:12.198317292+02:00" level=debug msg="[devmapper] removeDevice END(docker-202:1-1182467-HASH-init)" 
time="2016-03-29T14:29:12.198373996+02:00" level=debug msg="[devmapper] deactivateDevice END(HASH-init)" 
time="2016-03-29T14:29:12.198424823+02:00" level=debug msg="[devmapper] UnmountDevice(hash=HASH-init) END" 
time="2016-03-29T14:29:12.198821616+02:00" level=debug msg="activateDeviceIfNeeded(HASH)" 
time="2016-03-29T14:29:12.231530385+02:00" level=debug msg="[devmapper] UnmountDevice(hash=HASH)" 
time="2016-03-29T14:29:12.994368921+02:00" level=debug msg="[devmapper] Unmount(/var/XXXX/docker/devicemapper/mnt/HASH)" 
time="2016-03-29T14:29:13.001181907+02:00" level=debug msg="[devmapper] deactivateDevice(HASH)" 
time="2016-03-29T14:29:13.001298698+02:00" level=debug msg="[devmapper] removeDevice START(docker-202:1-1182467-HASH)" 
time="2016-03-29T14:29:13.008938191+02:00" level=debug msg="[devmapper] removeDevice END(docker-202:1-1182467-HASH)" 
time="2016-03-29T14:29:13.008991796+02:00" level=debug msg="[devmapper] deactivateDevice END(HASH)" 
time="2016-03-29T14:29:13.009055766+02:00" level=debug msg="[devmapper] UnmountDevice(hash=HASH) END" 
time="2016-03-29T14:29:13.012025705+02:00" level=info msg="POST /v1.19/containers/HASH/attach?stderr=1&stdout=1&stream=1" 
time="2016-03-29T14:29:13.015202739+02:00" level=info msg="POST /v1.19/containers/HASH/start" 
time="2016-03-29T14:29:13.471932242+02:00" level=debug msg="activateDeviceIfNeeded(HASH)" 
time="2016-03-29T14:29:14.509750363+02:00" level=debug msg="[devmapper] UnmountDevice(hash=HASH)" 
time="2016-03-29T14:29:18.106841833+02:00" level=debug msg="[devmapper] Unmount(/var/XXXX/docker/devicemapper/mnt/HASH)" 
time="2016-03-29T14:29:18.108516855+02:00" level=debug msg="[devmapper] deactivateDevice(HASH)" 
time="2016-03-29T14:29:18.108599137+02:00" level=debug msg="[devmapper] removeDevice START(docker-202:1-1182467-HASH)" 
time="2016-03-29T14:29:18.114042684+02:00" level=debug msg="[devmapper] removeDevice END(docker-202:1-1182467-HASH)" 
time="2016-03-29T14:29:18.114097305+02:00" level=debug msg="[devmapper] deactivateDevice END(HASH)" 
time="2016-03-29T14:29:18.114142994+02:00" level=debug msg="[devmapper] UnmountDevice(hash=HASH) END" 
time="2016-03-29T14:29:18.115658033+02:00" level=debug msg="[devmapper] UnmountDevice(hash=HASH)" 
time="2016-03-29T14:29:18.892828564+02:00" level=debug msg="[devmapper] UnmountDevice(hash=HASH) END" 
time="2016-03-29T14:29:18.892913082+02:00" level=error msg="Error unmounting device HASH: UnmountDevice: device not-mounted id HASH" 
time="2016-03-29T14:29:18.893117084+02:00" level=error msg="Handler for POST /containers/{name:.*}/start returned error: Cannot start container HASH: lstat /dev/dm-7: no such file or directory" 
time="2016-03-29T14:29:18.893190541+02:00" level=error msg="HTTP Error" err="Cannot start container HASH: lstat /dev/dm-7: no such file or directory" statusCode=404 
time="2016-03-29T14:29:18.896228952+02:00" level=info msg="DELETE /v1.19/containers/HASH?v=1" 
time="2016-03-29T14:29:20.046558280+02:00" level=debug msg="unregisterDevice(6273, HASH)" 
time="2016-03-29T14:29:20.051493777+02:00" level=debug msg="unregisterDevice(6271, HASH-init)" 
```

Thanks

**Comment by ramjothikumar** [comment]
I am seeing this intermittently as below.

```[root@host ~]# cat t
time="2016-05-03T20:37:26.158425221Z" level=info msg="POST /v1.21/build?buildargs=%7B%7D&cgroupparent=&cpuperiod=0&cpuquota=0&cpusetcpus=&cpusetmems=&cpushares=0&dockerfile=dockerfiles%2Fwheezy&memory=0&memswap=0&rm=1&t=collectd-test-image-wheezy&ulimits=null" 

time="2016-05-03T20:37:31.755976461Z" level=info msg="POST /v1.21/containers/create?name=collectd-test-wheezy-1462307846" 

time="2016-05-03T20:37:32.722346202Z" level=info msg="POST /v1.21/containers/d58f35e4b3773b6459ceac998de3f2376149cfb7cf08f945e557587d4661f93a/start" 

time="2016-05-03T20:37:33.780206459Z" level=warning msg="failed to cleanup ipc mounts:\nfailed to umount /storage/disk0/docker/containers/d58f35e4b3773b6459ceac998de3f2376149cfb7cf08f945e557587d4661f93a/shm: invalid argument\nfailed to umount /storage/disk0/docker/containers/d58f35e4b3773b6459ceac998de3f2376149cfb7cf08f945e557587d4661f93a/mqueue: invalid argument" 

time="2016-05-03T20:37:33.780273582Z" level=error msg="Error unmounting device d58f35e4b3773b6459ceac998de3f2376149cfb7cf08f945e557587d4661f93a: UnmountDevice: device not-mounted id d58f35e4b3773b6459ceac998de3f2376149cfb7cf08f945e557587d4661f93a" 

time="2016-05-03T20:37:33.780392014Z" level=error msg="Handler for POST /v1.21/containers/d58f35e4b3773b6459ceac998de3f2376149cfb7cf08f945e557587d4661f93a/start returned error: Cannot start container d58f35e4b3773b6459ceac998de3f2376149cfb7cf08f945e557587d4661f93a: lstat /dev/dm-5: no such file or directory" 

time="2016-05-03T20:37:33.780423222Z" level=error msg="HTTP Error" err="Cannot start container d58f35e4b3773b6459ceac998de3f2376149cfb7cf08f945e557587d4661f93a: lstat /dev/dm-5: no such file or directory" statusCode=500 

[root@host ~]# docker info
Containers: 8
Images: 581
Server Version: 1.9.1
Storage Driver: devicemapper
 Pool Name: docker-202:16-524291-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 107.4 GB
 Backing Filesystem: 
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 24.61 GB
 Data Space Total: 107.4 GB
 Data Space Available: 54.36 GB
 Metadata Space Used: 37.99 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.109 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /storage/disk0/docker/devicemapper/devicemapper/data
 Metadata loop file: /storage/disk0/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.93-RHEL7 (2015-01-28)
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.14.19-17.43.amzn1.x86_64
Operating System: Amazon Linux AMI 2014.09
CPUs: 8
Total Memory: 29.46 GiB
Name: ip-10-1-6-195.ec2.internal
ID: VXLJ:4I23:3M2X:XA5R:UZAI:GLU6:2TRB:5X3C:NAQL:6VXE:MVI6:PZV7
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled

[root@host ~]# docker version
Client:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5/1.9.1
 Built:        
 OS/Arch:      linux/amd64

Server:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5/1.9.1
 Built:        
 OS/Arch:      linux/amd64

[root@host ~]# cat /etc/*-release
NAME="Amazon Linux AMI"
VERSION="2014.09"
ID="amzn"
ID_LIKE="rhel fedora"
VERSION_ID="2014.09"
PRETTY_NAME="Amazon Linux AMI 2014.09"
ANSI_COLOR="0;33"
CPE_NAME="cpe:/o:amazon:linux:2014.09:ga"
HOME_URL="http://aws.amazon.com/amazon-linux-ami/"
Amazon Linux AMI release 2014.09```

**Comment by brucellino** [comment]
Can I just say that `--device=/dev/fuse --security-opt apparmor:unconfined` sounds  like :angel: :microphone: :notes: :guitar: I've been struggling for months.

**Comment by thaJeztah** [comment]
I have seen no recent reports of this issue, and i'm going to close, and lock this issue to prevent it from becoming a catch-all issue for possibly unrelated issues.

If you're still running into this, please open a new issue instead
